Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -219,17 +229,135 @@ namespace AzFramework
         }
     }
 
+    void XcbNativeWindow::DrawSplash()
+    {
+        const unsigned int bytesPerPixel = 4;
+        
+        typedef struct
+        {
+            const png_byte* data;
+            const png_size_t size;
+        } DataHandle;
+
+        typedef struct
+        {
+            const DataHandle data;
+            png_size_t offset;
+        } ReadDataHandle;
+
+        auto ReadDataFromInputStream = [](
+            png_structp png_ptr,
+            png_byte* raw_data,
+            png_size_t read_length)
+        {
+            ReadDataHandle* handle = (ReadDataHandle*)png_get_io_ptr(png_ptr);
+            const png_byte* png_src = handle->data.data + handle->offset;
+
+            memcpy(raw_data, png_src, read_length);
+            handle->offset += read_length;
+        };
+
+        png_structp png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
+        png_infop info_ptr = png_create_info_struct(png_ptr);
+
+        ReadDataHandle handle = {{ logo_png, logo_png_len},0};
+
+        png_set_read_fn(png_ptr, &handle, ReadDataFromInputStream);
+        png_read_info(png_ptr, info_ptr);
+
+        int width = png_get_image_width(png_ptr, info_ptr);
+        int height = png_get_image_height(png_ptr, info_ptr);
+
+        png_bytep row_pointers[height];
```

Comment:
Per the coding standard variable names are required to be [lowerCamelCase](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#variable-names).

Reply:
good to know :smile: